### ISSUE TITLE
"ZPublisher.HTTPRequest.HTTPRequest.get": docstring not conform with the code

### ISSUE THREAD

**Issue by d-maurer** [zopefoundation/Zope#629]
```
    def get(self, key, default=None, returnTaints=0, ...
            BASEmatch=re.compile('BASE(PATH)?([0-9]+)$').match,
            ):
        """Get a variable value

        Return a value for the required variable name.
        The value will be looked up from one of the request data
        categories. The search order is environment variables,
        other variables, form data, and then cookies.
        """
        other = self.other
        if key in other:
            if key == 'REQUEST':
                return self
            return other[key]
       ...
```
While the docstring says: "first `environ` then `other`", the code implements "first `other`".

**Comment by icemac** [comment]
A PR to fix the docstring is welcome or do you think that the docstring should be the correct one and the implementation should be fixed?

**Comment by d-maurer** [comment]
Michael Howitz wrote at 2019-5-16 23:19 -0700:
>A PR to fix the docstring is welcome

Planned - but I have currently more pressing (paid) work to do.

>or do you think that the docstring should be the correct one and the implementation should be fixed?

As `set` stores the set values in `other`, it is a good thing to
look there first. It would be difficult to understand if after
a `request.set(k, v)` `request.get(k)` would give something different from
`v`.

However, I have doubts to use `other` as a cache for `cookie`
and `form` lookups (see my other issue (--> "minor security risk")).

For this issue, I plan a PR as well (removing the caching -- even
though this might introduce a minor backward incompatibility).

**Comment by dataflake** [comment]
Fixed by #632
